lib/beaker/testmode_switcher/local_runner.rb

```diff
@@ -31,7 +31,12 @@ def execute_manifest(manifest, opts = {})
         cmd << "--debug" if opts[:debug]
         cmd << "--noop" if opts[:noop]
         cmd << "--trace" if opts[:trace]
-        use_local_shell(cmd.join(' '), opts)
+
+        res = use_local_shell(cmd.join(' '), opts)
+        exit_codes = get_acceptable_puppet_run_exit_codes(opts)
+        raise "Unacceptable exit code returned: #{res.exit_code}. Acceptable code(s): #{exit_codes.join(', ')}" unless exit_codes.include?(res.exit_code)
```

Comment:
This should be wrapped in an exception. See https://ruby-doc.org/core/Exception.html for ruby's recommendations on this.